File: TEMA3/python/menuCalc.py
def mostrar_menu():
    print("\n-----MENU DE OPERACIONES-----\n(1) Sumar\n(2) Restar\n(3) Multiplicar\n(4) Dividir\n(5) Salir")
 
def pedir_numeros():
    print("Presione ENTER para realizar la operacion")
    numeros = []
    while True:
        entrada = input("Ingresa un número: ")
        if entrada == "":
            break
        try:
            numeros.append(float(entrada))
        except ValueError:
            print("Entrada inválida.")
    return numeros

def sumar(nums):
    return sum(nums)

def restar(nums):
    resultado = nums[0]
    for n in nums[1:]:
        resultado -= n
    return resultado

def multiplicar(nums):
    resultado = 1
    for n in nums:
        resultado *= n
    return resultado

def dividir(nums):
    resultado = nums[0]
    for n in nums[1:]:
        if n == 0:
            return "No definida, no se puede dividir entre 0."
        resultado /= n
    return resultado

while True:
    mostrar_menu()
    opcion = input("Seleccione una opcion: ")
    if opcion == "5":
        print("Saliendo del programa...")
        break
    if opcion not in ["1", "2", "3", "4"]:
        print("Opción inválida.")
        continue
    numeros = pedir_numeros()
    if len(numeros) < 2:
        print("Debes ingresar al menos 2 números.")
        continue
    if opcion == "1": print(f"La suma es: {sumar(numeros)}")
    elif opcion == "2": print(f"La resta es: {restar(numeros)}")
    elif opcion == "3": print(f"La multiplicacion es: {multiplicar(numeros)}")
    elif opcion == "4": print(f"La division es: {dividir(numeros)}")

File: TEMA3/cpp/menuCalc.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

void mostrarMenu() {
    cout << "\n---MENU---\n"
         << "(1) Sumar\n"
         << "(2) Restar\n"
         << "(3) Multiplicar\n"
         << "(4) Dividir\n"
         << "(5) Salir\n";
}

vector<double> pedirNumeros() {
    cout << "ENTER vacío para terminar\n";
    vector<double> nums;
    string linea;
    while (true) {
        cout << "Número: ";
        getline(cin, linea);
        if (linea.empty()) break;
        try {
            nums.push_back(stod(linea));
        } catch (...) {
            cout << "Entrada inválida.\n";
        }
    }
    return nums;
}

double sumar(vector<double>& nums) {
    double res = 0;
    for (double n : nums) res += n;
    return res;
}

double restar(vector<double>& nums) {
    double res = nums[0];
    for (int i = 1; i < nums.size(); i++)
        res -= nums[i];
    return res;
}

double multiplicar(vector<double>& nums) {
    double res = 1;
    for (double n : nums) res *= n;
    return res;
}

bool dividir(vector<double>& nums, double& res) {
    res = nums[0];
    for (int i = 1; i < nums.size(); i++) {
        if (nums[i] == 0) return false;
        res /= nums[i];
    }
    return true;
}

int main() {
    string opcion;
    while (true) {
        mostrarMenu();
        cout << "Opción: ";
        getline(cin, opcion);
        if (opcion == "5") break;
        if (opcion != "1" && opcion != "2" &&
            opcion != "3" && opcion != "4") {
            cout << "Inválida.\n"; continue;
        }
        auto nums = pedirNumeros();
        if (nums.size() < 2) {
            cout << "Mínimo 2 números.\n"; continue;
        }
        double res;
        if (opcion == "1")
            cout << "Suma: " << sumar(nums) << "\n";
        else if (opcion == "2")
            cout << "Resta: " << restar(nums) << "\n";
        else if (opcion == "3")
            cout << "Multiplicación: "
                 << multiplicar(nums) << "\n";
        else if (opcion == "4") {
            if (!dividir(nums, res))
                cout << "No definida (div/0).\n";
            else cout << "División: " << res << "\n";
        }
    }
    return 0;
}